src/Client.php

```diff
@@ -259,6 +259,28 @@ protected function getForceIpv4Flag(ConnectionInterface $connection)
     }
 
     /**
+     * Extracts the value of the allow-selfsigned option from a given connection.
+     *
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @return boolean TRUE to allow self signed certificates, FALSE otherwise
+     */
+    protected function getAllowSelfSignedFlag(ConnectionInterface $connection)
+    {
+        return (boolean) $connection->getOption('allow-selfsigned') ?: false;
```

Comment:
I would use `allow-self-signed` instead of `allow-selfsigned` here, for consistency with the socket option being set.